scripts/globals/pets/Kupipi.lua

```diff
@@ -0,0 +1,283 @@
+	    local battletime = os.time()
+		local cureTime = mob:getLocalVar("cureTime")
+
+		if (battletime > cureTime + cureCooldown) then
+		    local party = player:getParty()
+            for _,member in ipairs(party) do
+                if (member:getHPP() <= 30) then
+					local spell = doEmergencyCure(mob)
+					if (spell > 0) then
+                        mob:castSpell(spell, member)
+					    mob:setLocalVar("cureTime",battletime)
+                        break
+					end
+                elseif (member:getHPP() <= 75) then
+					local spell = doCure(mob)
+					if (spell > 0) then
+                        mob:castSpell(spell, member)
+					    mob:setLocalVar("cureTime",battletime)
+                        break
+					end
+			    end
+			end
+			mob:setLocalVar("cureTime",battletime - 4)  -- If no member has low HP change global check to 8 seconds
+		end
+	end)
+
+	mob:addListener("TRUST_COMBAT_TICK", "KUPIPI_COMBAT_TICK" .. kupipi, function(mob, target)
+	    if (mob:getTP() > 1000) then
+		    weaponskill = doKupipiWeaponskill(mob)
+			mob:useMobAbility(weaponskill, mob)
+		end
+	end)
+end
+
+function doStatusRemoval(mob, player)
+	local mp = mob:getMP()
+	local lvl = mob:getMainLvl()
+	local spell = 0
+
+	if (player:hasStatusEffect(dsp.effect.POISON) and lvl >= 6 and mp >= 8) then
+	    spell = 14
+	elseif (player:hasStatusEffect(dsp.effect.PARALYSIS) and lvl >= 9 and mp >= 12) then
+	    spell = 15
+	elseif (player:hasStatusEffect(dsp.effect.BLINDNESS) and lvl >= 14 and mp >= 16) then
+	    spell = 16
+	elseif (player:hasStatusEffect(dsp.effect.SILENCE) and lvl >= 19 and mp >= 24) then
+	    spell = 17
+	elseif (player:hasStatusEffect(dsp.effect.CURSE_I) and lvl >= 29 and mp >= 30) then
+	    spell = 20
+	elseif (player:hasStatusEffectByFlag(dsp.effectFlag.ERASABLE) and lvl >= 32 and mp >= 18) then
+	    spell = 143
+	elseif (player:hasStatusEffect(dsp.effect.DISEASE) and lvl >= 34 and mp >= 20) then
+	    spell = 19
+	elseif (player:hasStatusEffect(dsp.effect.PETRIFICATION) and lvl >= 39 and mp >= 40) then
+	    spell = 18
+    end
+
+    return spell
+end
+
+function doKupipiWeaponskill(mob)
+    local wsList = {{43,164}, {1,163}}
+    local newWsList = {}
+	local maxws = 3 -- Maximum number of weaponskills to choose from randomly
+	local wscount = 0
+	local lvl = mob:getMainLvl()
+	local finalWS = 0
+
+	for i = 1, #wsList do
+		if (lvl >= wsList[i][1]) then
+			table.insert(newWsList, wscount + 1, wsList[i][2])
+			wscount = wscount + 1
+			if (wscount >= maxws) then
+				break
+			end
+		end
+	end
+
+	finalWS = newWsList[math.random(1,#newWsList)]
+	return finalWS
+end
+
+function doBuff(mob, player)
+    local proRaList = {{63,65,128}, {47,46,127}, {27,28,126}, {7,9,125}}
+    local proList = {{63,65,46}, {47,46,45}, {27,28,44}, {7,9,43}}
+    local shellRaList = {{68,75,133}, {57,56,132}, {37,37,131}, {17,18,130}}	
+    local shellList = {{68,75,51}, {57,56,50}, {37,37,49}, {17,18,48}}
+	local battletime = os.time()
```

Comment:
tabs threw off yer indents, you may want to get the editorconfig plugin for your code editor to automate this

Reply:
Thanks I thought I went thru the lua to check for spacing but I must have overwritten